src/pages/workspaces/workspace/WorkspaceMenu.ts

```diff
@@ -0,0 +1,172 @@
+import { icon } from '@terra-ui-packages/components';
+import { cond, DEFAULT } from '@terra-ui-packages/core-utils';
+import { Fragment, ReactNode } from 'react';
+import { h } from 'react-hyperscript-helpers';
+import { Clickable } from 'src/components/common';
+import { MenuButton } from 'src/components/MenuButton';
+import { makeMenuIcon, MenuTrigger } from 'src/components/PopupTrigger';
+import { useWorkspaceDetails } from 'src/components/workspace-utils';
+import { isOwner, WorkspacePolicy, WorkspaceWrapper as Workspace } from 'src/libs/workspace-utils';
+
+const isNameType = (o: WorkspaceInfo): o is DynamicWorkspaceInfo =>
+  'name' in o && typeof o.name === 'string' && 'namespace' in o && typeof o.namespace === 'string';
+
+type LoadedWorkspaceInfo = { canShare: boolean; isLocked: boolean; isOwner: boolean; workspaceLoaded: boolean };
+type DynamicWorkspaceInfo = { name: string; namespace: string };
+type WorkspaceInfo = DynamicWorkspaceInfo | LoadedWorkspaceInfo;
+
+interface WorkspaceMenuCallbacks {
+  onClone: () => void;
+  onShare: (policies?: WorkspacePolicy[]) => void;
+  onLock: () => void;
+  onDelete: () => void;
+  onLeave: () => void;
+}
+
+export interface WorkspaceMenuProps {
+  iconSize?: number;
+  popupLocation: unknown; // will error whenever PopupTrigger.js is converted to ts
```

Comment:
Hmmm, PopupTrigger has been converted

Reply:
It's been ported to `packages/components`, but the existing one in `src/components` hasn't changed.

Reply:
The old module now re-exports the TypeScript PopupTrigger from packages/components.